pkg/nsx/services/securitypolicy/cleanup.go

```diff
@@ -9,6 +9,36 @@ import (
 	"github.com/vmware-tanzu/nsx-operator/pkg/nsx/services/common"
 )
 
+// CleanupBeforeVPCDeletion cleans up share resources before VPC deletion to avoid dependency issues
+func (service *SecurityPolicyService) CleanupBeforeVPCDeletion(ctx context.Context) error {
```

Comment:
Having offline discussed with @timdengyun , we need to delete all SecurityPolicy resources before cleaning up the Shares, otherwise, the shares are failed to delete because the path shared in the NSX share resource is still consumed.

Reply:
added